src/Store.js

```diff
@@ -355,80 +385,28 @@ class Store {
    *     }
    *   })
    *
+   * NOTE: A broader RFC is in development to improve how we keep data in sync
+   * with the server. We likely will want to getMatchingRecords and getRecords
+   * to return null if nothing is found. However, this causes several regressions
+   * in portal we will need to address in a larger PR for mobx-async-store updates.
+   *
    * @method findAll
    * @param {String} type the type to find
    * @param {Object} options
    */
   findAll = (type, options = {}) => {
     const { fromServer, queryParams } = options
-
     if (fromServer === true) {
-      // If fromServer is true always fetch the data and return
-      return this.fetchAll(type, queryParams)
+      return this.fetchAll(type, options)
     } else if (fromServer === false) {
-      // If fromServer is false never fetch the data and return
       return this.getMatchingRecords(type, queryParams) || []
     } else {
-      return this.findOrFetchAll(type, queryParams) || []
-    }
-  }
-
-  /**
-   * @method findAndFetchAll
-   * @param {String} type the type to find
-   * @param {Object} options
-   * @return {Array}
-   */
-  findAndFetchAll = (type, options = {}) => {
-    const {
-      beforeFetch,
-      afterFetch,
-      beforeRefetch,
-      afterRefetch,
-      afterError,
-      queryParams
-    } = options
-
-    const records = this.getMatchingRecords(type, queryParams)
-
-    // NOTE: See note findOrFetchAll about this conditional logic.
-    if (records.length > 0) {
-      beforeRefetch && beforeRefetch(records)
-      this.fetchAll(type, queryParams)
-          .then((result) => afterRefetch && afterRefetch(result))
-          .catch((error) => afterError && afterError(error))
-    } else {
-      beforeFetch && beforeFetch(records)
-      this.fetchAll(type, queryParams)
-          .then((result) => afterFetch && afterFetch(result))
-          .catch((error) => afterError && afterError(error))
-    }
-
-    return records || []
-  }
-
-  /**
-   * returns cache if exists, returns promise if not
-   *
-   * @method findOrFetchAll
-   * @param {String} type record type
-   * @param {Object} queryParams will inform whether to return cached or fetch
-   */
```

Comment:
We weren't using findAndFetchAll anywhere but I saw Joe was implementing some kind of lazy loading. Should I keep this (do we plan to use this?)